client/extensions/woocommerce/state/sites/customers/selectors.js

```diff
@@ -0,0 +1,88 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+/**
+ * @param {Object} state Whole Redux state tree
+ * @param {String} searchTerm Search term to check
+ * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
+ * @return {boolean} Whether the search results for a given term have been successfully loaded from the server.
+ */
+export const isCustomerSearchLoaded = (
+	state,
+	searchTerm,
+	siteId = getSelectedSiteId( state )
+) => {
+	const inFlight = get(
+		state,
+		`extensions.woocommerce.sites[${ siteId }].customers.isSearching[${ searchTerm }]`
+	);
+	// Strict check because it could also be undefined.
+	return false === inFlight;
+};
+
+/**
+ * @param {Object} state Whole Redux state tree
+ * @param {String} searchTerm Search term to check
+ * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
+ * @return {boolean} Whether the search results for a given term are currently being retrieved from the server.
+ */
+export const isCustomerSearchLoading = (
+	state,
+	searchTerm,
+	siteId = getSelectedSiteId( state )
+) => {
+	const inFlight = get(
+		state,
+		`extensions.woocommerce.sites[${ siteId }].customers.isSearching[${ searchTerm }]`
+	);
+	// Strict check because it could also be undefined.
+	return true === inFlight;
+};
+
+/**
+ * @param {Object} state Whole Redux state tree
+ * @param {String} searchTerm Search term to check
+ * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
+ * @return {array} List of order notes
+ */
+export const getCustomerSearchResults = (
+	state,
+	searchTerm,
+	siteId = getSelectedSiteId( state )
+) => {
+	if ( ! isCustomerSearchLoaded( state, searchTerm, siteId ) ) {
+		return [];
+	}
+
+	const customers = get( state, `extensions.woocommerce.sites[${ siteId }].customers.items`, {} );
+	const customerIdsForTerm = get(
+		state,
+		`extensions.woocommerce.sites[${ siteId }].customers.queries[${ searchTerm }]`,
+		[]
+	);
+	return customerIdsForTerm.map( id => customers[ id ] );
```

Comment:
Do we need to ensure `customers[ id ]` exists to prevent an `undefined` from being returned in this array. Unlikely to happen I suppose.

Reply:
It shouldn't happen by way of how the reducer works, but I've added a catch just in case.